File: python/producer.py
from confluent_kafka import Producer
import sys

producer = Producer({
  'bootstrap.servers' : 'localhost:9092',
  'client.id'         : 'kafka-demo',
  'security.protocol' : 'SASL_PLAINTEXT',
  'sasl.mechanism'    : 'PLAIN',
  'sasl.username'     : 'test',
  'sasl.password'     : 'test-secret'
})

clusterMetaData = producer.list_topics()
print(clusterMetaData.topics)

def delivery_callback(err, msg):
  if err:
    sys.stderr.write('%% Message failed delivery: %s\n' % err)
  else:
    sys.stderr.write('%% Message delivered to %s [%d] @ %d\n' % (msg.topic(), msg.partition(), msg.offset()))

producer.poll(0);

producer.produce(topic = "test-topic", value = "hello world!", key = "key1", partition = 0, callback = delivery_callback)
producer.produce(topic = "test-topic", value = "hey hey!", key = "key2", partition = 0, callback = delivery_callback)
producer.produce(topic = "test-topic", value = "hi hi!", key = "key3", partition = 1, callback = delivery_callback)

producer.flush();

File: go/producer.go
package main

import (
	"fmt"
	"os"
	"time"

	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
)

func main() {
	p, err := kafka.NewProducer(&kafka.ConfigMap{
		"bootstrap.servers"	: "localhost:9092",
		"client.id"         : "kafka-demo",
		"security.protocol" : "SASL_PLAINTEXT",
		"sasl.mechanism"    : "PLAIN",
		"sasl.username"     : "test",
		"sasl.password"     : "test-secret",
	})

	if err != nil {
		fmt.Printf("Failed to create Admin client: %s\n", err)
		os.Exit(1)
	}

	topic := new(string)
	*topic = "test-topic"

	metaData, err := p.GetMetadata(topic, false, 60 * 1000)

	if err != nil {
		fmt.Printf("Failed to get topic metadata: %v\n", err)
		os.Exit(1)
	}

	topicMetaData := metaData.Topics[*topic]
	fmt.Printf("%s\n", topicMetaData.Topic)

	go func() {
		for e := range p.Events() {
			switch ev := e.(type) {
			case *kafka.Message:
				m := ev
				if m.TopicPartition.Error != nil {
					fmt.Printf("Delivery failed: %v\n", m.TopicPartition.Error)
				} else {
					fmt.Printf("Delivered message to topic %s [%d] at offset %v\n",
						*m.TopicPartition.Topic, m.TopicPartition.Partition, m.TopicPartition.Offset)
				}
			case kafka.Error:
				fmt.Printf("Error: %v\n", ev)
			default:
				fmt.Printf("Ignored event: %s\n", ev)
			}
		}
	}()

	msgcnt := 0
	for msgcnt < 3 {
		value := fmt.Sprintf("Producer example, message #%d", msgcnt)

		err = p.Produce(&kafka.Message{
			TopicPartition: kafka.TopicPartition{Topic: topic, Partition: kafka.PartitionAny},
			Value					: []byte(value),
			Key						: []byte("key1"),
		}, nil)

		if err != nil {
			if err.(kafka.Error).Code() == kafka.ErrQueueFull {
				time.Sleep(time.Second)
				continue
			}
			fmt.Printf("Failed to produce message: %v\n", err)
		}
		msgcnt++
	}

	for p.Flush(5000) > 0 {
		fmt.Print("Still waiting to flush outstanding messages\n")
	}

	p.Close();
}